### PR TITLE
sign_up

### PR DESCRIPTION
**Pull request by MenBui**
==> Title: Sign up 

==> Redmine: https://edu-redmine.sun-asterisk.vn/issues/19255

==> Attachment File:
Screenshot from 2019-11-22 09-55-30

### CODE REVIEW

**Comment by MenBui** [comment]
ready

**Comment by MenBui** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
```
Comment (phucmh-1356) [inline app/controllers/users_controller.rb]: k check đk k tìm thấy @user à em

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    else
+      flash[:danger] = t "create_false"
+      render :new
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :name, :email, :sex, :birth,
```
Comment (phucmh-1356) [inline app/controllers/users_controller.rb]: viết 1 hằng số USER_PARAMS = %i(name email sex birth phone address identity_card password password_confirmation).freeze trong model user rồi gọi ra ở đây bằng cách User::USER_PARAMS

File: app/models/category.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 class Category < ApplicationRecord
   has_many :books, dependent: :destroy
-  has_many :childrens, class_name: Category.name, foreign_key: :parent_id,
-  belongs_to :parent, class_name: Category.name, optional: true
+  has_many :childrens, class_name: Category.name, foreign_key: :parent_id
+  belongs_to :parent, class_name: Category.name, optional: true,
           dependent: :destroy, inverse_of: :parent
```
Comment (phucmh-1356) [inline app/models/category.rb]: indent 2 spaces

File: app/models/user.rb [hunk 4]
```diff
@@ -2,4 +2,21 @@ class User < ApplicationRecord
...
+  validates :email, presence: true, length: {maximum: Settings.maximum_email},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  validates :password, presence: true, length: {minimum:
+    Settings.minimum_password}
+  validates :phone, presence: true, length: {maximum:
+    Settings.maximum_phone}
+  validates :address, length: {maximum: Settings.maximum_address}
+  validates :identity_card, presence: true, length: {maximum:
+    Settings.maximum_identity_card}
+  enum sex: {male: Settings.enum_active, female: Settings.enum_archived}
```
Comment (phucmh-1356) [inline app/models/user.rb]: other thì sao em

**Comment by MenBui** [comment]
ready

**Pull request merged**